sphinx_book_theme/layout.html

```diff
@@ -20,7 +20,27 @@
 <main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
     {% block docs_body %}
     {% include "topbar.html" %}
-    <div id="main-content" class="row">
+    <div id="jb-print-docs-body" class="onlyprint">
+        <h1>{{ pagetitle }}</h1>
+        <!-- Table of contents -->
+        <div id="main-content" class="row onlyprint">
+            <div class="col-12 col-md-12 pl-md-5 pr-md-5">
+            <div>
+                {% set page_toc = generate_toc_html() %}
+                {%- if page_toc | length >= 1 %}
+                <div>
+                    <h2> {{ translate(theme_toc_title) }} </h2>
+                </div>
+                <nav aria-label="Page">
+                    {{ page_toc }}
+                </nav>
+                {%- endif %}
+            </div>
+            {{ super() }}
+            </div>
+        </div>
+    </div>
+    <div id="main-content" class="row noprint">
```

Comment:
I'm confused - wouldn't this mean that the whole `#main-content` div doesn't show up when printing?

Reply:
This #main-content won't, but this one will https://github.com/executablebooks/sphinx-book-theme/pull/438/files#diff-6cf877d43a0155d9e3706cb22f143f326d4283458bfdcac93017b197c4f02e2bR27 . Looking at this now, i should rename the id of `onlyprint` one to something else, as there are two `main-content` ids at one time in this page.

Reply:
or maybe convert `main-content` to a `class` name instead of `id`.

Reply:
hmmm - it seems reasonable to use a unique ID for the print-only header. That seems like a smaller change than converting it from an ID to a class, IMO

Reply:
Fair enough. Changed them to individual ids.